src/deroff.rs

```diff
@@ -273,6 +273,51 @@ impl Deroffer {
             _ => return None,
         })
     }
+    //     def skip_char(self, amt=1):
+    //         self.s = self.s[amt:]
+    fn skip_char<'a>(&self, s: &'a str, amount: Option<usize>) -> &'a str {
+        let amount = amount.unwrap_or(1);
+        s.get(amount..).unwrap_or("")
+    }
+
+    //     def skip_leading_whitespace(self):
+    //         self.s = self.s.lstrip()
+    fn skip_leading_whitespace<'a>(&self, s: &'a str) -> &'a str {
+        s.trim_start()
+    }
+    //     def str_at(self, idx):
+    //         return self.s[idx:idx+1]
+    fn str_at<'a>(s: &'a str, idx: usize) -> &'a str {
+        // This doesn't fully mirror the python version functionality
+        // It could be implemented with `str::chars` method on a utf-8 string
+        // should we need that
+        s.get(idx..idx + 1).unwrap_or("")
+    }
+
+    //     def is_white(self, idx):
+    //         # Note this returns false for empty strings (idx >= len(self.s))
+    //         return self.s[idx:idx+1].isspace()
+    fn is_white<'a>(s: &'a str, idx: usize) -> bool {
+        match Self::str_at(s, idx) {
+            "" => false,
+            c => c.chars().all(|c| c.is_ascii_whitespace()),
```

Comment:
Do we want to account for unicode and use `is_whitespace()`

Reply:
Good question, that also affects https://github.com/rust-dc/fish-manpage-completions/pull/33/files#diff-5b0804df3203e0b4fc4f245618d61d53R291

Have we decided yet if we'll parse everything, potentially lossy to utf?